accounting/src/main/java/io/skai/accounting/repository/impl/OrderRepositoryImpl.java

```diff
@@ -6,22 +6,35 @@
 import org.jooq.DSLContext;
 import org.springframework.stereotype.Repository;
 
+import static io.skai.accounting.enums.OrderStatus.NEW;
 import static io.skai.accounting.jooq.Tables.ORDER;
+import static io.skai.accounting.jooq.Tables.ORDER_STATUS;
+
 @Repository
 @RequiredArgsConstructor
 public class OrderRepositoryImpl implements OrderRepository {
     private final DSLContext dslContext;
+
     @Override
     public Order create(Order order) {
-        return dslContext.insertInto(ORDER)
-                .set(ORDER.CLIENT_ID, order.getClientId())
-                .set(ORDER.MANAGER_ID, order.getManagerId())
-                .set(ORDER.DEFECT, order.getDefect())
-                .set(ORDER.SERIAL_NUMBER, order.getSerialNumber())
-                .set(ORDER.MODEL_ID, order.getModelId())
-                .returning()
-                .fetchOptional()
-                .orElseThrow()
-                .into(Order.class);
+        return dslContext.transactionResult(trx -> {
+            Order savedOrder = dslContext.insertInto(ORDER)
+                    .set(ORDER.CLIENT_ID, order.getClientId())
+                    .set(ORDER.MANAGER_ID, order.getManagerId())
+                    .set(ORDER.DEFECT, order.getDefect())
+                    .set(ORDER.SERIAL_NUMBER, order.getSerialNumber())
+                    .set(ORDER.MODEL_ID, order.getModelId())
+                    .returning()
+                    .fetchOptional()
+                    .orElseThrow()
+                    .into(Order.class);
+
+            dslContext.insertInto(ORDER_STATUS)
```

Comment:
If transaction to order status table fail, what fallback will be? It's a bad approach to do distributed commit in such way, please change it to avoid non consistent cases.